c7n/resources/vpc.py

```diff
@@ -878,12 +879,41 @@ class UsedSecurityGroup(SGUsage):
     """
     schema = type_schema('used')
 
+    instance_owner_id_key = 'c7n:InstanceOwnerIds'
+    interface_type_key = 'c7n:InterfaceTypes'
+
+    def _set_eni_attributes(self):
+        enis = []
+        for nic in self.nics:
+            if nic['Status'] == 'in-use':
+                instance_owner_id = nic['Attachment']['InstanceOwnerId']
+            else:
+                instance_owner_id = ''
+            interface_type = nic.get('InterfaceType')
+            for g in nic['Groups']:
+                enis.append({'GroupId': g['GroupId'],
+                             'InstanceOwnerId': instance_owner_id,
+                             'InterfaceType': interface_type})
+        return enis
+
     def process(self, resources, event=None):
         used = self.scan_groups()
         unused = [
             r for r in resources
             if r['GroupId'] not in used and 'VpcId' in r]
         unused = {g['GroupId'] for g in self.filter_peered_refs(unused)}
+        enis = self._set_eni_attributes()
+        for r in resources:
+            r[self.instance_owner_id_key] = []
+            r[self.interface_type_key] = []
+            for i in range(1, len(enis)):
```

Comment:
`for eni in enis` might be a bit simpler and more readable here, and would let you replace `enis[i]` with `eni` inside the loop

Reply:
Fixed. `for eni in enis` is simpler indeed. It was a leftover from an old logic that I tried to do a different way.